pkg/util/tar_utils.go

```diff
@@ -106,21 +107,52 @@ func unpackTar(tr *tar.Reader, path string) error {
 				return err
 			}
 			currFile.Close()
-		}
+		case tar.TypeSymlink:
+			// It's possible we end up creating files that can't be overwritten based on their permissions.
+			// Explicitly delete an existing file before continuing.
+			if _, err := os.Stat(target); !os.IsNotExist(err) {
+				logrus.Debugf("Removing %s to create symlink.", target)
+				if err := walkAndRemove(target); err != nil {
+					logrus.Debugf("Unable to remove %s: %s", target, err)
+				}
+			}
 
+			if err = os.Symlink(header.Linkname, target); err != nil {
+				logrus.Errorf("Failed to create symlink between %s and %s: %s", header.Linkname, target, err)
+			}
+		}
 	}
 	return nil
 }
 
+func walkAndRemove(p string) error {
+	return filepath.Walk(p, func(path string, info os.FileInfo, err error) error {
+		if e := os.Chmod(path, 0777); e != nil {
+			logrus.Errorf("Error updating file permissions on %s before removing for symlink creation", path)
+			return e
+		}
+		return os.RemoveAll(path)
+	})
+}
+
+func checkWhitelist(target string, whitelist []string) bool {
```

Comment:
I think you want something more precise than prefix matching. Something like /foo would block /foobar when it should only block /foo/bar